### PR TITLE
feature/allow-list > setting domain to be whitelisted if confirmed on signed transaction and checking if is whitelisted

### PR DESCRIPTION
**Pull request by jeesunikim**
**Summary:**
- renamed `whitelist` to `allowList`
- checking to see if the domain the user is on is within `ALLOWLIST_ID` on localStorage.
- `<GrantAccess/>` to display a warning message at all time

### CODE REVIEW

File: extension/src/background/messageListener/lyraApiMessageListener.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import { publicKeySelector } from "background/ducks/session";
 
 import { responseQueue, transactionQueue } from "./popupMessageListener";
 
-const WHITELIST_ID = "whitelist";
+const ALLOWLIST_ID = "allowlist";
```
Comment (piyalbasu) [inline extension/src/background/messageListener/lyraApiMessageListener.ts:21]: I noticed this issue the other day, we redeclare this const in both `lyraApiMessageListener` and `popupMessageListener`. Why don't we make a `background/constants` file and store this there

File: extension/src/background/messageListener/lyraApiMessageListener.ts [hunk 2]
```diff
@@ -65,18 +65,29 @@ export const lyraApiMessageListener = (
...
+    const { transactionXdr } = request;
     const transaction = StellarSdk.TransactionBuilder.fromXDR(
       transactionXdr,
       StellarSdk.Networks[NETWORK],
     );
 
     const { tab } = sender;
+    const tabUrl = tab?.url ? tab.url : "";
+
+    const sanitizedUrl = removeQueryParam(tabUrl);
```
Comment (piyalbasu) [inline extension/src/background/messageListener/lyraApiMessageListener.ts:79]: I actually really liked what you did with `punycode` in `popup`. Maybe we should create a helper method that takes a url and runs it through `getUrlHostname` and `punycode.toASCII` to get just the punycode domain and store that in the allowlist

That would also solve the issue of a different path on the same domain triggering the warning message again

Comment (jeesunikim) [reply extension/src/background/messageListener/lyraApiMessageListener.ts]: Updated @piyalbasu

File: extension/src/helpers/urls.ts [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
   return decodedSearchParam ? JSON.parse(decodedSearchParam) : {};
 };
+
 export const getUrlHostname = (url: string) => {
   const u = new URL(url);
   return u.hostname;
 };
+
+export const getPunycodedDomain = (url: string) => {
+  return punycode.toASCII(url);
```
Comment (piyalbasu) [inline extension/src/helpers/urls.ts:18]: oooh interesting Github Actions warning. Yeah, let's fix this error

File: extension/src/background/messageListener/lyraApiMessageListener.ts [hunk 4]
```diff
@@ -65,18 +67,30 @@ export const lyraApiMessageListener = (
...
+    const tabUrl = tab?.url ? tab.url : "";
+    const domain = getUrlHostname(tabUrl);
+    const punycodedDomain = getPunycodedDomain(domain);
+
+    const allowListStr = localStorage.getItem(ALLOWLIST_ID) || "";
+    const allowList = allowListStr.split(",");
+
+    if (allowList.includes(punycodedDomain)) {
+      isDomainListedAllowed = true;
+    }
```
Comment (piyalbasu) [inline extension/src/background/messageListener/lyraApiMessageListener.ts:88]: super small, but it's a little less error prone to do:
```
isDomainListedAllowed = allowList.includes(punycodedDomain);
```

Comment (jeesunikim) [reply extension/src/background/messageListener/lyraApiMessageListener.ts]: Updated, @piyalbasu

**Review by piyalbasu — APPROVED** [review]

**Pull request merged**